### DISCUSSION TITLE
how about adding extractor： ValidatedJson<T>

### DISCUSSION THREAD

**Discussion by zishon** [tokio-rs/axum#2806] [Ideas]
how about adding extractor： ValidatedJson<T> 
which T derive  validator::Validate
and the behavior should be
1. if the header does not have Content-Type: application/json，then content-type error
2. if the request body can't be deserialized，then not a valid json string error
3. if the payload ( deserialized into T) validate fails  （payload.validate()）, then validate error
4. if no error happens, then go ahead.

**Comment by mladedav** [comment]
Do you mean having something like `axum-garde` inside `axum-extra`?

**Reply by zishon** [reply]
Yes, sort of.
